### ISSUE TITLE
Navigate + queryParams and useRedirect

### ISSUE THREAD

**Issue by sdirosa** [kyeotic/raviger#1]
First and foremost, thanks for this. I just started playing around with raviger and I really like it so far!

I've a couple of questions.

For context; we have a react application with a few pathnames and a bunch of parameters as query strings. Some of them are optional, other obligatory. Some get hydrated as entities, others are just UI settings. 

Most importantly we share some of them between pages, which means that we adjust them when a user switches from a route to another. Eg: some are preserved, other are removed or altered. 

Questions:

1. Do you plan on supporting passing an object with queryParams to the `navigate` as hookerouter does? At the moment I use `navigate` in combination with `useParams` and do something like: 

```
navigate(path + '?' + (URLSearchParams(queryParams).toString()))
```

Which feels a bit odd :confused: 

2. Any plans on adding a (use)Redirect functionality like hookrouter has?
If not, do you plan to expose the router context so we may extend on that and build it ourselves?
If that's also not an option, what do you think would be the best way to handle redirects with raviger?

Thanks!

**Comment by kyeotic** [comment]
Yes, I would love to make `navigate` smarter, especially around query strings. What do you think about it acceting a [Location-like](https://developer.mozilla.org/en-US/docs/Web/API/Location) object?

**Comment by kyeotic** [comment]
2) Yes, I can add that. Would you mind opening a second issue to track this request?

**Comment by kyeotic** [comment]
1. I think this makes sense, but I am not sure how to square it with the custom query serialization that the `useQueryParams` hooks provides. I need to figure this out first. Perhaps you could define it as part of the routing context?

You can easily create such a function in your app and use it instead of the raviger one.

```javascript
import {navigate as nav} from 'raviger'
export function navigate (path, query, replace) {
  return nav(path + '?' + (URLSearchParams(query).toString()), replace)
}
```

**Comment by kyeotic** [comment]
I guess custom serialization can still be easily accomplished using the above method, so providing the standard serialization is still a net gain. I'll do that.

**Comment by sdirosa** [comment]
> 1. Yes, I can add that. Would you mind opening a second issue to track this request?

Sure! And thanks for the quick reply. Here you go: https://github.com/kyeotic/raviger/issues/2

**Comment by sdirosa** [comment]
> I guess custom serialization can still be easily accomplished using the above method, so providing the standard serialization is still a net gain. I'll do that.

Cool, looking forward to test it out :+1:

**Comment by kyeotic** [comment]
@sdirosa I added #3. Feel free to add a review, if not I will merge it in later today

**Comment by kyeotic** [comment]
published under `0.3.3`
